metrics/ci_integrations/test/firestore/adapter/firestore_storage_client_adapter_test.dart

```diff
@@ -0,0 +1,137 @@
+import 'package:ci_integration/firestore/deserializer/build_data_deserializer.dart';
+import 'package:ci_integration/firestore/adapter/firestore_storage_client_adapter.dart';
+import 'package:firedart/firedart.dart';
+import 'package:metrics_core/metrics_core.dart';
+import 'package:mockito/mockito.dart';
+import 'package:test/test.dart';
+import 'package:grpc/grpc.dart';
+
+void main() {
+  group('FirestoreStorageClientAdapter', () {
+    const projectId = 'projectId';
+    const testDocumentId = 'documentId';
+    final buildDataTestJson = {
+      'buildNumber': 1,
+      'startedAt': DateTime.now(),
+      'buildStatus': BuildStatus.failed.toString(),
+      'duration': 100,
+      'workflowName': 'testWorkflowName',
+      'url': 'testUrl',
+      'coverage': 0.1,
+    };
+    final documentReferenceMock = _DocumentReferenceMock();
+    final collectionAddBuildsMock = _CollectionReferenceMockAddBuilds(
+      projectId: projectId,
+      documentReferenceMock: documentReferenceMock,
+    );
+    final collectionFetchBuildsMock = _CollectionReferenceMockFetchBuild(
+      projectId: projectId,
+    );
+    final firestoreMock =
+        _FirestoreMock(collectionAddBuildsMock, collectionFetchBuildsMock);
+    final mockAdapter = FirestoreStorageClientAdapter(firestoreMock);
+
```

Comment:
Do we need to reset all mocks using the `reset()` function from the mockito package to ensure that tests are running independently?